flocker/node/agents/blockdevice.py

```diff
@@ -1490,6 +1517,12 @@ def discover_state(self, node_state):
             NodeState(
                 uuid=self.node_uuid,
                 hostname=self.hostname,
+                # FLOC-1874 The sizes of the datasets for these manifestations
+                # will be the size of the actual storage-driver-supplied
+                # volume.  This may not be the size requested in the
+                # configuration.  File a follow-up issue for fixing this if it
+                # matters.  Fixing this requires having access to the
```

Comment:
Insofar as:

1. `calculate_changes` knows about fact that underlying storage size may not quite match, and
2. Lying about underlying storage size is quite difficult in some backends,

I think reporting accurately is the thing to do.